src/commands/list/list.js

```diff
@@ -1,28 +1,26 @@
 const { SlashCommandBuilder } = require('@discordjs/builders');
 const { prepareSongTitle } = require('@utils/formatString');
-const { MIN_PREV_TRACKS_TO_SHOW, MAX_NEXT_TRACKS_TO_SHOW } = require('@consts');
+const useList = require('@hooks/useList');
 
 module.exports = {
 	data: new SlashCommandBuilder()
 		.setName('list')
 		.setDescription('Get a list of songs from the current queue'),
 	async execute({ client, interaction }) {
-
-		const queue = client.player.queues.get(interaction.guildId);
-
-		if(!queue) {
-			return await interaction.reply('There is no queue');
+		const queue = client.player.queues.get(interaction.guild.id);
+		const [listState] = useList(queue);
+	
+		if (!queue) {
+			return new Error('Queue error')
 		}
-
+	
 		const currentTrack = queue.currentTrack || queue.history.tracks.data[0];
-
-		const prevTracksToShowNum = queue.currentTrack ? MIN_PREV_TRACKS_TO_SHOW : MIN_PREV_TRACKS_TO_SHOW + 1;
-
-		const prevTracks = queue.history.tracks.data.slice(queue.currentTrack ? 0 : 1, prevTracksToShowNum).reverse().map(prepareSongTitle).join('\n\t');
-		const nextTracks = queue.tracks.data.slice(0, MAX_NEXT_TRACKS_TO_SHOW).map(prepareSongTitle).join('\n\t');
-
-		let displayedQueue = '\t' + prevTracks + '\n> ' + prepareSongTitle(currentTrack) + '\n\t' + nextTracks;
-
+		const prevTracksToShowNum = queue.currentTrack ? listState.MIN_PREV_TRACKS : listState.MIN_PREV_TRACKS + 1;
+		const prevTracks = queue.history.tracks.data.slice(queue.currentTrack ? 0 : 1, prevTracksToShowNum).reverse().map(prepareSongTitle);
+		const nextTracks = queue.tracks.data.slice(0, listState.MAX_NEXT_TRACKS).map(prepareSongTitle);
+	
+		let displayedQueue = '\t' + prevTracks.join('\n\t') + '\n> ' + prepareSongTitle(currentTrack) + '\n\t' + nextTracks.join('\n\t');
+	
```

Comment:
all this logic must be a part of a class TrackList. TrackList::GetList for example